sdk/eventhub/azure-eventhub/azure/eventhub/exceptions.py

```diff
@@ -161,10 +161,10 @@ def _handle_exception(
         name = closable._container_id  # pylint: disable=protected-access
     if isinstance(exception, KeyboardInterrupt):  # pylint:disable=no-else-raise
         _LOGGER.info("%r stops due to keyboard interrupt", name)
-        closable.close()
+        closable._close_connection()  # pylint:disable=protected-access
         raise exception
     elif isinstance(exception, EventHubError):
-        closable.close()
+        closable._close_connection()  # pylint:disable=protected-access
```

Comment:
I would stick to calling `close` instead of `_close_connection_async`.

My concern is that we'll have shared connection in the future, if EventHubError is a link/session level error, then connection shouldn't be closed.

Reply:
Per our discussion, use _close_handler instead of _close_connection for EventHubError here.